hazelcast/src/main/resources/hazelcast-config-3.5.xsd

```diff
@@ -655,6 +656,20 @@
             </xs:simpleType>
         </xs:attribute>
     </xs:complexType>
+    <xs:complexType name="ringbuffer">
+        <xs:all>
+            <xs:element name="capacity" type="xs:unsignedInt" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="retention-seconds" type="xs:unsignedInt" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="backup-count" type="backup-count" minOccurs="0" maxOccurs="1" default="1"/>
+            <xs:element name="async-backup-count" type="backup-count" minOccurs="0" maxOccurs="1" default="0"/>
+            <xs:element name="in-memory-format" type="in-memory-format" minOccurs="0" maxOccurs="1" default="BINARY"/>
```

Comment:
I'm not completely sure what you mean. 

Should the XMLConfigBuilder not catch all exceptions, including IllegalArgumentException, and wrap it with an InvalidConfigurationException? 

Because most of the methods throws IllegalArgument or NPE when the arguments are not correct. 

So I'm a bit lost here. Can you elaborate?